File: src/Test/ArrayOpsFFI.js
export const runArrayOpsFFI = function(limit) {
  let n = Math.floor(limit);
  let arr = [];
  for (let i = 1; i <= n; i++) {
    arr.push(i);
  }
  
  let evens = arr.filter(function(x) { return x % 2 === 0; });
  
  let sum = evens.reduce(function(acc, x) { return acc + x; }, 0);
  
  return sum;
};

File: src/Test/ArrayOpsFFI.java
    private static Integer[] rangeFromOne(int end) {
        int size = Math.toIntExact(Math.abs((long) end - 1) + 1);
        Integer[] result = new Integer[size];
        int direction = end >= 1 ? 1 : -1;
        for (int i = 0; i < size; i++) result[i] = 1 + direction * i;
        return result;
    }

    public static final java.util.function.Function<Object, Object> runArrayOpsFFI = Bench.nativeBenchmark(input -> {
        Integer[] values = rangeFromOne((Integer) input);
        Integer[] evens = java.util.Arrays.stream(values).filter(value -> value % 2 == 0).toArray(Integer[]::new);
        return java.util.Arrays.stream(evens).reduce(0, (sum, value) -> sum + value);
    });
